### PR TITLE
Add IL tests for cases where Span is not allowed

### PR DESCRIPTION
**Pull request by kouvel**
Fixes https://github.com/dotnet/coreclr/issues/8516

### CODE REVIEW

**Comment by jkotas** [comment]
I have finished writing a similar (smaller) set of tests for https://github.com/dotnet/coreclr/pull/15745 literally 10 minutes ago...

File: tests/src/baseservices/collections/span/SpanDisallowedTestsIL.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
+﻿using System;
```
Comment (jkotas) [inline tests/src/baseservices/collections/span/SpanDisallowedTestsIL.cs]: License headers

**Comment by jkotas** [comment]
Not sure whether it is worth keeping both sets from both my and this PR ... there seems to be quite a bit of overlap.

LGTM otherwise.

**Comment by kouvel** [comment]
Maybe I can merge them

File: tests/src/baseservices/collections/span/SpanDisallowedTests.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
```
Comment (ahsonkhan) [inline tests/src/baseservices/collections/span/SpanDisallowedTests.cs]: same here (license header)

**Pull request closed without merging**